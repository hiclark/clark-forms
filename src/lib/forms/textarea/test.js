import React from 'react';

import Textarea from './';

describe('<Textarea />', () => {
  let wrapper;
  let textarea;
  let label;

  beforeEach(() => {
    const props = {
      type: 'textarea',
      name: 'Name',
      label: 'Label',
      index: 1,
      placeholder: 'Placeholder Text',
      showError: false,
      meta: {
        touched: false,
        error: false,
      },
    };
    wrapper = shallow(<Textarea {...props} />);
    textarea = wrapper.find('Field');
    label = wrapper.find('textarea__Label');
  });

  describe('<Textarea /> field renders props correctly', () => {
    test('render name prop', () => {
      expect(textarea.prop('name')).toMatch('Name');
    });

    test('render placeholder prop', () => {
      expect(textarea.prop('placeholder')).toMatch('Placeholder Text');
    });

    test('render type prop', () => {
      expect(textarea.prop('type')).toMatch('textarea');
    });

    test('render index prop', () => {
      expect(textarea.prop('index')).toEqual(1);
    });

    test('render showError prop', () => {
      expect(textarea.prop('showError')).toBeFalsy();
    });
  });

  describe('<Textarea /> label renders correctly', () => {
    test('render label text', () => {
      expect(label.render().text()).toEqual('Label');
    });
  });

  describe('<Textarea /> renders error component correctly', () => {
    test('error component snapshot test', () => {
      expect(textarea).toMatchSnapshot();
      wrapper.setProps({
        meta: {
          touched: true,
          error: true,
        },
      });
      expect(textarea).toMatchSnapshot();
    });
  });

  test('<Textarea /> snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
