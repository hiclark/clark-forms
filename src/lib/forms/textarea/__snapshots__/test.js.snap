// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Textarea /> <Textarea /> snapshot test 1`] = `
<div>
  <Label
    disabled={false}
    label="Label"
    name="Name"
  />
  <Field
    component={[Function]}
    index={1}
    label="Label"
    meta={
      Object {
        "error": false,
        "touched": false,
      }
    }
    name="Name"
    placeholder="Placeholder Text"
    showError={false}
    type="textarea"
  />
</div>
`;

exports[`<Textarea /> renders error component correctly error component snapshot test 1`] = `
<Field
  component={[Function]}
  index={1}
  label="Label"
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  name="Name"
  placeholder="Placeholder Text"
  showError={false}
  type="textarea"
/>
`;

exports[`<Textarea /> renders error component correctly error component snapshot test 2`] = `
<Field
  component={[Function]}
  index={1}
  label="Label"
  meta={
    Object {
      "error": false,
      "touched": false,
    }
  }
  name="Name"
  placeholder="Placeholder Text"
  showError={false}
  type="textarea"
/>
`;
